src/components/MediaHandler/MediaHandler.js

```diff
@@ -16,6 +17,7 @@ export class MediaHandler extends React.Component {
       const location = await FileInterface.saveMedia();
       mediaChanged(location);
     } catch {}
+    this.setState({buttonText: "Change Media"});
```

Comment:
I think we can remove this line? obtaining a media does not guarantee a render

Reply:
Maybe it should be moved to inside the try statement?

Reply:
there is a then block inside load media which updates the state if there is a (relatively) successful load of the file. you can put it in there

Reply:
Turns out I had it in there as well.